doc/design/obs-call-home.md

```diff
@@ -0,0 +1,165 @@
+
+- Volume Size Metrics (Provisioned Size)
+  - maximum size
+  - minimum size
+  - average size
+  - percentiles: 50%, 75%, 90%
+
+  This information is state-based, which can be taken from the control-plane (via the public REST API) as required (once per 24 hours won’t cause excessive load on the control-plane)
+
+- Pool Size Metrics (Total Capacity)
+  - maximum capacity
+  - minimum capacity
+  - average capacity
+  - percentiles: 50%, 75%, 90%
+
+  This information is state-based, which can be taken from the control-plane (via the public REST API) as required (once per 24 hours won’t cause excessive load on the control-plane)
+
+- Health Metrics
+
+  To be defined in the "Health Metrics" section, below
+
+- Churn Metrics
+
+  For both the past 60 minute and past 24hr periods:
+  - count of volumes created
+  - count of volumes deleted
+  - count of replicas created
+  - count of replicas deleted
+
+  These require time series calculation and storage, to be performed by the "Statistics Module" (see below)
+
+## Where are we sending it to?
+
+- Remote "Health Reports" S3 bucket: <https://openebs.phonehome.datacore.com/openebs/report>
+- It's sent as an encrypted [JSON] document
+
+## When / How frequently do we send the information?
+
+- On first start of the telemetry pod
+- Every 24 hours thereafter (default)
+
+## Solution Components
+
+A modular architecture, decoupling the call-home transmission from the actual collection of data and from the (user) Health metrics exporter (Prometheus scrapes).
+
+This makes it very simple for the user to disable call-home report sending, which is a requirement as some users may not be comfortable with sending
+this information out.
+
+### Event Bus
+
+Any call-home required events (e.g. volume/replica creation and deletion) are collected from the event bus.
+
+These events should be raised within the appropriate code path, data-plane / control-plane.
+
+The Metrics / Statistics modules subscribe to these event/message topics and process the messages, updating their internal metrics and statistical values accordingly.
+
+The event bus doesn't need to be persistent since an event loss does not affect the products main functionality, though it may impair its traceability.
+You can read more about events [here](./events.md).
+
+### Events (control plane and/or data plane)
+
+There are creation/deletion events for various resources, eg:
+
+- create/delete volumes
+- create/delete replicas
+
+As well as edge-triggered events such as:
+
+- Volume degraded
+- Replica faulted
+
+### Health Statistics Component
+
+A component whose role is to store time series information for call-home and health metrics:
+
+- metrics which can’t be simply derived from current state of the control-plane or data-plane components
+  - i.e. number of replicas created since installation
```

Comment:
Is indentation off here or it should just have been in the previous line?

Reply:
It's intended this way